tonic/src/codec/compression.rs

```diff
@@ -157,8 +174,6 @@ impl CompressionEncoding {
         }
     }
 
-    #[allow(missing_docs)]
-    #[cfg(any(feature = "gzip", feature = "zstd"))]
     pub(crate) fn as_str(&self) -> &'static str {
         match self {
```

Comment:
I think derefenrencing here can solve the non-exhaustive error.

```suggestion
        match *self {
```

Reply:
wouldn't that have more to do with the `#[non_exhaustive]` on `CompressionEncoding`?

Reply:
The match can't see through the `&` to realize that the match is otherwise exhaustive and in the same crate as the `#[non_exhaustive]` enum is defined in. By dereferencing the `Copy` type, the exhaustiveness check can succeed.

Reply:
Changed the function to take `self`.